packages/by-name/kata/kata-runtime/0016-genpolicy-support-mount-propagation-and-ro-mounts.patch

```diff
@@ -0,0 +1,68 @@
+From 12799d175ad019bc6cdeaef6a6316a081ae56510 Mon Sep 17 00:00:00 2001
+From: Markus Rudy <mr@edgeless.systems>
+Date: Mon, 23 Sep 2024 20:58:54 +0200
+Subject: [PATCH 16/16] genpolicy: support mount propagation and ro-mounts
+
+---
+ src/tools/genpolicy/rules.rego               |  6 ++++--
+ src/tools/genpolicy/src/mount_and_storage.rs | 14 ++++++++++++--
+ 2 files changed, 16 insertions(+), 4 deletions(-)
+
+diff --git a/src/tools/genpolicy/rules.rego b/src/tools/genpolicy/rules.rego
+index 685109ba6..b95508dbc 100644
+--- a/src/tools/genpolicy/rules.rego
++++ b/src/tools/genpolicy/rules.rego
+@@ -105,7 +105,8 @@ allow_create_container_input {
+     count(i_linux.GIDMappings) == 0
+     count(i_linux.MountLabel) == 0
+     count(i_linux.Resources.Devices) == 0
+-    count(i_linux.RootfsPropagation) == 0
++    # TODO(burgerdev): is it harmful to always allow RootfsPropagation?
++    # count(i_linux.RootfsPropagation) == 0
+     count(i_linux.UIDMappings) == 0
+     is_null(i_linux.IntelRdt)
+     is_null(i_linux.Resources.BlockIO)
+@@ -522,7 +523,8 @@ allow_by_bundle_or_sandbox_id(p_container, i_oci, i_storages) {
+     i_root := i_oci.Root.Path
+     p_root_pattern1 := p_oci.Root.Path
+     p_root_pattern2 := replace(p_root_pattern1, "$(cpath)", policy_data.common.cpath)
+-    p_root_pattern3 := replace(p_root_pattern2, "$(bundle-id)", "([0-9a-f]{64})")
++    # bundle-id can be either a container name or a container ID.
++    p_root_pattern3 := replace(p_root_pattern2, "$(bundle-id)", concat("", ["(", policy_data.common.dns_label, ")"]))
```

Comment:
Yes, related to the PR: the rootfs path seems to contain the container name rather than the id if bidirectional propagation is on. Don't ask me why. Note that the check I'm replacing here is tighter than the check in upstream/msft, which reads the bundle path from a host-provided annotation.

No, unrelated to this patch and should have gone into 0008.

Reply:
> the rootfs path seems to contain the container name rather than the id if bidirectional propagation is on

Do we want to add this to the comment or patch description?

Reply:
I was looking for the code where this happens, but did not find it. Then I tried to reproduce the issue, but it did not occur. Reverting the change now.